p2p/kademlia/node.go

```diff
@@ -2,132 +2,99 @@ package kademlia
 
 import (
 	"bytes"
+	"fmt"
 	"math/big"
-	"net"
-	"strconv"
+	"strings"
+
+	"github.com/jbenet/go-base58"
 )
 
-// NetworkNode is the over-the-wire representation of a node
-type NetworkNode struct {
-	// ID is a 20 byte unique identifier
+// Node is the over-the-wire representation of a node
+type Node struct {
+	// id is a 20 byte unique identifier
 	ID []byte
 
-	// IP is the IPv4 address of the node
-	IP net.IP
+	// ip address of the node
+	IP string
 
-	// Port is the port of the node
+	// port of the node
 	Port int
 }
 
-// node represents a node in the network locally
-// a separate struct due to the fact that we may want to add some metadata
-// here later such as RTT, or LastSeen time
-type node struct {
-	*NetworkNode
+func (s *Node) String() string {
+	return fmt.Sprintf("%v-%v:%d", base58.Encode(s.ID), s.IP, s.Port)
 }
 
-// NewNetworkNode creates a new NetworkNode for bootstrapping
-func NewNetworkNode(ip string, port string) *NetworkNode {
-	p, _ := strconv.Atoi(port)
-	return &NetworkNode{
-		IP:   net.ParseIP(ip),
-		Port: p,
+// NewNode returns a new node for bootstrapping
+func NewNode(ip string, port int) *Node {
+	return &Node{
```

Comment:
where to set ID of node?

Reply:
this function is only used for bootstrap, i think it's better to remove it